packages/jsdom-crawler/src/internals/jsdom-crawler.ts

```diff
@@ -0,0 +1,1023 @@
+     * {
+     *     error: Error,
+     *     request: Request,
+     *     session: Session,
+     *     $: Cheerio,
+     *     body: String|Buffer,
+     *     json: Object,
+     *     contentType: Object,
+     *     response: Object,
+     *     proxyInfo: ProxyInfo,
+     *     crawler: CheerioCrawler,
+     * }
+     * ```
+     * where the {@link Request} instance corresponds to the failed request, and the `Error` instance
+     * represents the last error thrown during processing of the request.
+     *
+     * See [source code](https://github.com/apify/apify-js/blob/master/src/crawlers/cheerio_crawler.js#L13)
+     * for the default implementation of this function.
+     *
+     * @deprecated `handleFailedRequestFunction` has been renamed to `failedRequestHandler` and will be removed in a future version.
+     */
+    handleFailedRequestFunction?: CheerioFailedRequestHandler<JSONData>;
+
+    /**
+     * Async functions that are sequentially evaluated before the navigation. Good for setting additional cookies
+     * or browser properties before navigation. The function accepts two parameters, `crawlingContext` and `gotOptions`,
+     * which are passed to the `requestAsBrowser()` function the crawler calls to navigate.
+     * Example:
+     * ```
+     * preNavigationHooks: [
+     *     async (crawlingContext, gotOptions) => {
+     *         // ...
+     *     },
+     * ]
+     * ```
+     */
+    preNavigationHooks?: CheerioHook<JSONData>[];
+
+    /**
+     * Async functions that are sequentially evaluated after the navigation. Good for checking if the navigation was successful.
+     * The function accepts `crawlingContext` as the only parameter.
+     * Example:
+     * ```
+     * postNavigationHooks: [
+     *     async (crawlingContext) => {
+     *         // ...
+     *     },
+     * ]
+     * ```
+     */
+    postNavigationHooks?: CheerioHook<JSONData>[];
+
+    /**
+     * An array of [MIME types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Complete_list_of_MIME_types)
+     * you want the crawler to load and process. By default, only `text/html` and `application/xhtml+xml` MIME types are supported.
+     */
+    additionalMimeTypes?: string[];
+
+    /**
+     * By default `CheerioCrawler` will extract correct encoding from the HTTP response headers.
+     * Sadly, there are some websites which use invalid headers. Those are encoded using the UTF-8 encoding.
+     * If those sites actually use a different encoding, the response will be corrupted. You can use
+     * `suggestResponseEncoding` to fall back to a certain encoding, if you know that your target website uses it.
+     * To force a certain encoding, disregarding the response headers, use {@link CheerioCrawlerOptions.forceResponseEncoding}
+     * ```
+     * // Will fall back to windows-1250 encoding if none found
+     * suggestResponseEncoding: 'windows-1250'
+     * ```
+     */
+    suggestResponseEncoding?: string;
```

Comment:
I'm in favor of removing this option completely and doing what Chrome does. Having this as an option just adds complexity.